app/controllers/PaymentScheduleController.scala

```diff
@@ -45,25 +44,29 @@ class PaymentScheduleController @Inject() (
 
   implicit val localDateOrdering: Ordering[LocalDate] = _ compareTo _
 
-  val checkPaymentSchedule: Action[AnyContent] = as.eligibleJourneyAction.async { implicit request =>
+  val checkPaymentSchedule: Action[AnyContent] = as.eligibleJourneyAction { implicit request =>
     request.journey match {
-      case j: Journey.BeforeSelectedPaymentPlan =>
-        logErrorAndRouteToDefaultPageF(j)
+      case _: Journey.BeforeSelectedPaymentPlan =>
+        JourneyLogger.warn("Not enough information in session to show check payment plan page. " +
+          "Redirecting to missing info page")
+        Redirect(routes.MissingInfoController.missingInfo)
 
       case j: Journey.AfterSelectedPaymentPlan =>
         val upfrontPaymentAnswers =
           j.into[Journey.AfterUpfrontPaymentAnswers].transform.upfrontPaymentAnswers
         val paymentDay =
           j.into[Journey.AfterEnteredDayOfMonth].transform.dayOfMonth
 
-        finalStateCheckF(j, Future.successful(Ok(paymentSchedulePage(upfrontPaymentAnswers, paymentDay, j.selectedPaymentPlan))))
+        finalStateCheck(j, Ok(paymentSchedulePage(upfrontPaymentAnswers, paymentDay, j.selectedPaymentPlan)))
     }
   }
 
   val checkPaymentScheduleSubmit: Action[AnyContent] = as.eligibleJourneyAction.async { implicit request =>
     request.journey match {
-      case j: Journey.BeforeSelectedPaymentPlan =>
-        logErrorAndRouteToDefaultPageF(j)
+      case _: Journey.BeforeSelectedPaymentPlan =>
+        JourneyLogger.warn("Not enough information in session to show the check payment schedule page. " +
```

Comment:
Could we extract this warn into a def in JourneyLogger and just pass in the page as a string maybe, saves having the same thing over and over

Reply:
just pushed up a change for this - it's quite specific so I wasn't sure if `JourneyLogger` was the best place to factor it out. I stuck something in the companion object of `MissingInfoController`. What do you think?

Reply:
Yeah that sounds sensible